aoc/src/main/scala/AoC2015Day06.scala

```diff
@@ -0,0 +1,60 @@
+object AoC2015Day06 extends App:
+    import scala.io.Source
+    import scala.util.matching.Regex
+
+    case class Instruction(instructionType: String, bottomLeftX: Int, bottomLeftY: Int, topRightX: Int, topRightY: Int)
+
+    val instructions: List[Instruction] = 
+        Source.fromResource("2015/InputDay06.txt")
+        .getLines
+        .toList
+        .map(
+            str => str match
+                // turn on/turn off
+                case s"turn ${instructionType} ${bottomLeftX},${bottomLeftY} through ${topRightX},${topRightY}"
+                    => Instruction(instructionType, bottomLeftX.toInt, bottomLeftY.toInt, topRightX.toInt, topRightY.toInt)
+                // toggle
+                case s"${instructionType} ${bottomLeftX},${bottomLeftY} through ${topRightX},${topRightY}"
+                    => Instruction(instructionType, bottomLeftX.toInt, bottomLeftY.toInt, topRightX.toInt, topRightY.toInt)
+                case _ => throw new IllegalArgumentException
+        )        
+
+    val lightsArrayInit: List[List[Int]] = List.fill(1000)(List.fill(1000)(0))
+
+    def computeTotalLights(lightsArray: List[List[Int]], instrHandler: (Int, Instruction) => Int): Int = 
```

Comment:
nice handler - good way to factor out the differences between part 1 and 2

(im a bit against using `0` and `1` to indicate lights being on or off in part 1 though - you could try to abstract over the light type, using an `Boolean` for part 1 and an `Int` for part 2 for learning purposes)